samples/dotnetcore/image-transfer/DotNetTransfer/DotNetTransfer/Options.cs

```diff
@@ -0,0 +1,124 @@
+﻿using Microsoft.Azure.Management.ResourceManager.Fluent;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace DotNetTransfer
+{
+    public class Options
+    {
+        public string Environment { get; set; }
+        public string TenantId { get; set; }
+        public string MIClientId { get; set; }
+        public string SPClientId { get; set; }
+        public string SPClientSecret { get; set; }
+        public string SubscriptionId { get; set; }
+        public PipelineProperties ExportPipeline { get; set; }
+        public PipelineProperties ImportPipeline { get; set; }
+        public PipelineRunProperties ExportPipelineRun { get; set; }
+
+        public AzureEnvironment AzureEnvironment
+        {
+            get
+            {
+                return string.IsNullOrWhiteSpace(Environment)
+                    ? AzureEnvironment.AzureGlobalCloud
+                    : AzureEnvironment.FromName(Environment);
+            }
+        }
+
+        public void Validate()
+        {
+            if (string.IsNullOrWhiteSpace(TenantId))
+            {
+                throw new ArgumentNullException(nameof(TenantId));
+            }
+
+            if (string.IsNullOrWhiteSpace(MIClientId)
+                && (string.IsNullOrWhiteSpace(SPClientId) || string.IsNullOrWhiteSpace(SPClientSecret)))
+            {
+                throw new ArgumentNullException($"Missing {nameof(MIClientId)} or {nameof(SPClientId)}/{nameof(SPClientSecret)}");
+            }
+
+            if (string.IsNullOrWhiteSpace(SubscriptionId))
+            {
+                throw new ArgumentNullException(nameof(SubscriptionId));
+            }
+
+            ExportPipeline.Validate();
+            ImportPipeline.Validate();
+        }
+    }
+
+    public class PipelineProperties
```

Comment:
Sorry about the maybe asking the obvious here. Doesn't the SDK have the pipeline properties already that we can use without having to redefine the whole set of properties again. I would expect we just wrote a validate extension method on the ExportPipeline/ImportPipeline types.

Reply:
Also if these are needed could you split them up into respective files.

Reply:
These were just added to store the pipeline configs in appsettings.json to simplify for the user. We construct the resource requests for them with these properties, such as making the identity property or pipelineRun request. I will organize these classes into separate files.

Reply:
See update